### PR TITLE
Dockerization and CI

### PR DESCRIPTION
**Pull request by ramilexe**

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Run docker build
+        run: make docker-build
+      - name: Push to Docker Hub
+        uses: docker/build-push-action@v1
+        with:
+          username: vulcanizejenkins
+          password: ${{ secrets.VULCANIZEJENKINS_PAT }}
+          repository: vulcanizejenkins/eth-header-sync
```
Comment (AFDudley) [inline .github/workflows/publish.yaml]: The repo should be vulcanize/eth-header-sync

Comment (ramilexe) [reply .github/workflows/publish.yaml]: @AFDudley In Docker Hub our name is vulcanizejenkins. Could you give access to `vulcanize` account?

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    types: [published]
+jobs:
+  push_to_registries:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Run docker build
+        run: make docker-build
```
Comment (AFDudley) [inline .github/workflows/publish.yaml]: Maybe we want to keep the runs from build.yml in github and just pull them from there and push them to docker upon release?

File: Makefile [hunk 3]
```diff
@@ -93,3 +93,8 @@ checkdbvars:
 migrate: $(GOOSE) checkdbvars
 	$(GOOSE) -dir db/migrations postgres "$(CONNECT_STRING)" up
 	pg_dump -O -s $(CONNECT_STRING) > db/schema.sql
+
+## Build docker image
+.PHONY: docker-build
+docker-build:
+	docker build -t vulcanizejenkins/eth-header-sync -f dockerfiles/Dockerfile .
```
Comment (AFDudley) [inline Makefile]: vulcanize/eth-header-sync

**Comment by ramilexe** [comment]
Issue https://github.com/vulcanize/eth-header-sync/issues/5

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (ramilexe) [reply .github/workflows/publish.yaml]: Unfortunately, GitHub Actions doesn't support direct communications between workflow. There is a workaround for that (store docker image as an artifact on flow 1 and then download it on flow 2) but it looks ugly for me

Comment (AFDudley) [reply .github/workflows/publish.yaml]: I rather store the docker image than build it twice for no reason, it's also cheaper.

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (AFDudley) [reply .github/workflows/publish.yaml]: vulcanizejenkins already has access to the vulcanize account, please test to confirm.

**Review by i-norden — COMMENTED** [review]
Only thing I can think to add is to go ahead and move the contents of dockerfiles/ up to root

**Comment by ramilexe** [comment]
> Only thing I can think to add is to go ahead and move the contents of dockerfiles/ up to root

done

**Review by i-norden — APPROVED** [review]

**Pull request merged**